docs/control-structures.rst

```diff
@@ -324,12 +324,14 @@ In the following example, we show how ``throw`` can be used to easily revert an
 
 Currently, there are six situations, where exceptions happen automatically in Solidity:
 
-1. If you access an array beyond its length (i.e. ``x[i]`` where ``i >= x.length``).
-2. If a function called via a message call does not finish properly (i.e. it runs out of gas or throws an exception itself).
-3. If a non-existent function on a library is called or Ether is sent to a library.
-4. If you divide or modulo by zero (e.g. ``5 / 0`` or ``23 % 0``).
-5. If you perform an external function call targeting a contract that contains no code.
-6. If a contract-creation call using the ``new`` keyword fails.
+1. If you access an array on a too large or negative index (i.e. ``x[i]`` where ``i >= x.length`` or ``i < 0``).
+2. If you access a fixed-length bytes on a too large or negative index.
+3. If a function called via a message call does not finish properly (i.e. it runs out of gas, has no matching function, or throws an exception itself), except when a low level operation ``call``, ``send``, ``delegatecall`` or ``callcode`` is used.
+4. If a non-existent function on a library is called or Ether is sent to a library.
```

Comment:
Hm, you seem to have removed the ``payable`` reference?

Perhaps it is easier to split into two:
- If Ether is sent to a contract function without the ``payable`` modifier (including the constructor and the fallback function).
- If a non-existent function on a library is called.

Ether can be sent to a library because we have no way to distinguish between call and delegatecall from a callee perspective. @chriseth correct me if I'm wrong.